cfe_ts_crc.c

```diff
@@ -125,30 +127,43 @@ int main(int argc, char **argv)
     if (fd < 0)
     {
         printf("\ncfe_ts_crc error: can't open input file!\n");
-        exit(0);
+        printf("%s\n", strerror(errno));
+        exit(1);
     }
     /* seek past the number of bytes requested */
-    lseek(fd, skipSize, SEEK_SET);
+    offsetReturn = lseek(fd, skipSize, SEEK_SET);
+    if (offsetReturn != skipSize)
+    {
+        printf("\ncfe_ts_crc error: lseek failed!\n");
+        printf("%s\n", strerror(errno));
+        exit(1);
+    }
 
     /* read the input file 100 bytes at a time */
-    while (done == 0)
+    do
     {
         readSize = read(fd, buffer, 100);
-        fileCRC  = CalculateCRC(buffer, readSize, fileCRC);
+        if (readSize < 0)
+        {
+            printf("\ncfe_ts_crc error: file read failed!\n");
+            printf("%s\n", strerror(errno));
+            exit(1);
+        }
+        fileCRC = CalculateCRC(buffer, readSize, fileCRC);
         fileSize += readSize;
-        if (readSize != 100)
-            done = 1;
-    }
+    } while (readSize != 100);
+
     /* print the size/CRC results */
-    printf("\nTable File Name:            %s\nTable Size:                 %d Bytes\nExpected TS Validation CRC: "
+    printf("\nTable File Name:            %s\nTable Size:                 %ld Bytes\nExpected TS Validation CRC: "
            "0x%08X\n\n",
            argv[1], fileSize, fileCRC);
 
     /* Close file and check*/
     if (close(fd) != 0)
     {
         printf("\nerror: Cannot close file!\n");
-        exit(0);
+        printf("%s\n", strerror(errno));
+        exit(1);
     }
 
     return (fileCRC);
```

Comment:
I just noticed this one too!  It should not return the CRC as the return code - it should return 0 if successful.  The CRC value makes no sense to a shell.

Reply:
You sure about that, because the whole point of this if to get a CRC

https://github.com/nasa/tblCRCTool/blob/b02864ba56b12e00ab152225e3e8f9d6c039d48c/cfe_ts_crc.c#L21-L37

Reply:
It reports it, doesn't need to return it to the shell.  If it really needs to be stored it could go to a file or similar... doesn't make sense as an exit code though (and defeats error checking when scripted).

Reply:
Right - the return value from `main` is system-defined where `EXIT_SUCCESS` (0) should be returned if the program completed successfully.  A caller could never get the CRC via this value. 

For instance someone might reasonably call this tool from a makefile that builds an update package for their target.  The nonzero exit status will be interpreted by `make` as a failure of the tool and it will stop the build.  (similar to what happens if a .c file fails to compile with errors).

On success, all programs should return with an exit code of 0 - that's the unix way, and has been for decades.

Reply:
fixed